torch/_subclasses/fake_tensor.py

```diff
@@ -564,6 +571,9 @@ def index_put_(fake_mode, func, *args, **kwargs):
     _, new_kwargs = normalize_function(
         func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
     )
+    check_same_device(
+        new_kwargs["input"], new_kwargs["values"], allow_cpu_scalar_tensors=True
+    )
```

Comment:
I don't think this is right -- it looks like we allow values to be a CUDA scalar Tensor as well:
```
x = torch.tensor([1., 2.])
y = torch.tensor(2., device='cuda')
idx = torch.tensor(0)
torch.index_put(x, y, idx)
```

Reply:
looks like a mistake...

Reply:
Is there a different example you can give ? this doesn't run
```
x = torch.tensor([1., 2.])
y = torch.tensor(2., device='cuda')
idx = torch.tensor(0)
torch.index_put(x, y, idx)
>>> TypeError: index_put(): argument 'indices' (position 2) must be tuple of Tensors, not Tensor
```

Reply:
my bad, good catch!